materialize-boilerplate/validate.go

```diff
@@ -207,7 +207,22 @@ func (v Validator) validateMatchesExistingBinding(
 
 		// Build constraints for new projections of the binding.
 		if _, ok := constraints[p.Field]; !ok {
-			constraints[p.Field] = v.c.NewConstraints(&p, deltaUpdates)
+			c := v.c.NewConstraints(&p, deltaUpdates)
+
+			// Continue to recommended any optional fields that were included in a prior spec's
```

Comment:
You say:

> But, it will help to preserve continuity for "legacy" specs if
the definition of recommended fields changes for a materialization and there is a change in the
materialized resource. 

I'm a little tired, but: is this check worth the code? It seems like it could be trivially omitted and :shrug: but it may have come about due to an acute problem you saw.

Reply:
The described scenario in the comment is a bit of a stretch. A stronger motivation is for systems like dynamodb where we allow field selection, but the destination system has only a limited concept of a "column" (only the keys for dynamodb). In that case I think its useful to recommend fields that were previously selected, even if their "as new" constraints are optional, based on the field selection from the prior spec.

Reply:
Oh! Okay, that's a _great_ reason for doing this, but it didn't come through for me at all. Would you please update the comment to give that example?